### DISCUSSION TITLE
alpha: how to delete fields?

### DISCUSSION THREAD

**Discussion by peterpeterparker** [firebase/firebase-js-sdk#4513] [Q&A]
What's the correct way to delete a field with the new alpha version?

Previously I used:

```
myDoc.data.aField = firebase.firestore.FieldValue.delete()

await firestore.collection(`/myCollection`).doc(myDoc.id).set(myDoc.data, {merge: true});
```

With alpha I tried:

```
import {deleteField} from 'firebase/firestore';
myDoc.data.aField = deleteField();

const dbLite = getFirestoreLite(firebaseApp);
await setDoc(doc(collection(dbLite, `/myCollection`), myDoc.id), myDoc.data, {merge: true});
```

but get an error

> FirebaseError: [code=invalid-argument]: Function setDoc() called with invalid data. Unsupported field value: a custom object (found in field attributes.style in document myCollection/slBKP...)

**Comment by schmidt-sebastian** [comment]
I think this should work. I will try to reproduce and report back.

**Reply by schmidt-sebastian** [reply]
You seem to be using Firestore Lite, but `deleteField` is from "firebase/firestore". Unfortunately, "firebase/firestore" and "firebase/firestore/lite" do not yet share any code, so any `instanceof` checks will fail across the SDK boundary. Your snipper works if you import `deleteField` from "firebase/firestore/lite".

**Reply by peterpeterparker** [reply]
Nice thanks! Indeed it works out that was my mistake. I have to say, regarding design, this kinds of mistake can happens quite quickly, specially if you are using both "firebase/firestore" and "firebase/firestore/lite" in the same app.

**Reply by schmidt-sebastian** [reply]
Yes, we have talked about bundling these SDKs in a way that would make this more seamless. At the very least, we should have a better error message here.

**Reply by peterpeterparker** [reply]
Ah cool, sounds good 👍
